torch/distributed/_sharding_spec/api.py

```diff
@@ -41,6 +41,34 @@ class ShardingSpec(PlacementSpec):
     pass
 
 
+@dataclass
+class ReshardingSpec(object):
+    """
+    This is a wrapper of PlacementSpec that defines the new placement as being
+    resharded across multiple devices or aggregate into a local tensor.
+
+    The semantics of how a tensor is partitioned is inline with
+    :meth:`torch.chunk`, where ``dim`` in torch.chunk corresponds to the
+    specified ``dim`` and ``chunks`` in torch.chunk is the number of elements
+    in the placement specified.
+
+    Args:
+        spec (ShardingSpec):
+            The spec contains the new dimension to reshard on and placement.
+        output_local_tensors(bool):
+            Whether or not to output the local shards of the ShardedTensor as
+            a tensor.
+    """
+    spec: ShardingSpec
+    output_local_tensors: bool = False
+
+    def __post_init__(self):
+        if not isinstance(self.spec, ShardingSpec):
```

Comment:
Probably also check `ShardingSpec` is instance of `ChunkShardingSpec` since we will only support that for now.